lib/providers.go

```diff
@@ -62,16 +67,98 @@ func (m *MockDataSourceReader) UnmatchedCalls() []cty.Value {
 	return uc
 }
 
+// GetPluginFolder tries to compute the user plugin folder for the current user.
+// For windows: %APPDATA%/terraform.d/plugins
+// For linux: ~/terraform.d/plugins
+func GetPluginFolder() (string, error) {
+	homeDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+
+	if runtime.GOOS == "windows" {
+		return filepath.FromSlash(fmt.Sprintf("%s/AppData/Roaming/terraform.d/plugins", homeDir)), nil
+	}
+
+	return filepath.FromSlash(fmt.Sprintf("%s/.terraform.d/plugins", homeDir)), nil
+}
+
+// parseProviderValues retrives the values for hostname, namespace and provider name from the path.
+func parseProviderValues(provMeta discovery.PluginMeta) (*addrs.Provider, error) {
+	parts := strings.Split(filepath.ToSlash(provMeta.Path), "/")
+
+	partCount := len(parts)
+
+	// overall the path does not have to be that long for tf12
+	if partCount < 3 {
+		return nil, fmt.Errorf("No valid provider path: %s", provMeta.Path)
+	}
+
+	if parts[partCount-3] == "plugins" {
+		// plugins folder was initialized by terraform <=0.12
+		return &addrs.Provider{
+			Hostname:  addrs.DefaultRegistryHost,
+			//Namespace: addrs.LegacyProviderNamespace,
+			// Fake this, because terraform 13 will look for this namespace by default
+			Namespace: "hashicorp",
+			Type:      provMeta.Name,
+		}, nil
```

Comment:
```suggestion
		return addrs.NewDefaultProvider(provMeta.Name), nil
```
It will do the same thing but hide the "hack" in a nice official method :slightly_smiling_face:

Reply:
I will change it accordingly but keep the comment about the hack. Hiding hacks is bad. ;)